File: RedSheet.DbEntities/Models/CultureContry.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("CultureContries", Schema = "dbo")]
	[Serializable]
	public partial class CultureContry  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int CultureCountryId { get; set; } 
 
	[Required]
	public bool IsEgalitarian { get; set; } 
 
	[Required]
	public bool IsIndividualistic { get; set; } 
 
	[Required]
	public bool IsMonochronic { get; set; } 
 
	[Required]
	public bool IsShortTerm { get; set; } 
 
	[RelationshipTableAttribue("Countries","dbo")]
	//Relationships
	public int CountryId { get; set; } 
 
	[ForeignKey("CountryId")]
	public Country Country { get; set; } 

	public CultureContry()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/culture-contry.ts
import { required, maxLength, range } from '@rx/annotations';
import { Country,  } from './'
export class CultureContry {
    constructor(cultureContry?: CultureContry )  {
        let properties = [ "cultureCountryId", "isEgalitarian", "isIndividualistic", "isMonochronic", "isShortTerm", "countryId",];
        for (let property of properties)
            if (cultureContry && cultureContry[property])
                this[property] = cultureContry[property];
    }
 
	cultureCountryId : number =   0 ;
 
	isEgalitarian : boolean = false ;
 
	isIndividualistic : boolean = false ;
 
	isMonochronic : boolean = false ;
 
	isShortTerm : boolean = false ;
 
	countryId : number =   undefined;
	country : Country  ;
}
